osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerUserModSelectOverlay.cs

```diff
@@ -0,0 +1,210 @@
+        /// Responds to changes in the local user's style to take on the server-side state.
+        /// </summary>
+        private void onUserStyleChanged(MultiplayerRoomUser user) => Scheduler.Add(() =>
+        {
+            if (user.Equals(client.LocalUser))
+                updateFromOnlineState();
+        });
+
+        /// <summary>
+        /// Responds to changes in the local user's mods to take on the server-side state.
+        /// </summary>
+        private void onUserModsChanged(MultiplayerRoomUser user) => Scheduler.Add(() =>
+        {
+            if (user.Equals(client.LocalUser))
+                updateFromOnlineState();
+        });
+
+        /// <summary>
+        /// Responds to changes in the current playlist item to take on the server-side state,
+        /// potentially also changing the visible mod panels depending on the new item state.
+        /// </summary>
+        private void onPlaylistItemChanged(MultiplayerPlaylistItem item) => Scheduler.Add(() =>
+        {
+            if (item.ID == client.Room?.Settings.PlaylistItemId)
+                updateFromOnlineState();
+        });
+
+        /// <summary>
+        /// Responds to changes in the current playlist item to take on the server-side state,
+        /// potentially also changing the visible mod panels depending on the new item state.
+        /// </summary>
+        private void onSettingsChanged(MultiplayerRoomSettings settings) => Scheduler.Add(() =>
+        {
+            if (settings.PlaylistItemId != lastPlaylistItemId)
+            {
+                updateFromOnlineState();
+                lastPlaylistItemId = settings.PlaylistItemId;
+            }
+        });
+
+        /// <summary>
+        /// Tracks changes to the local selected mods to notify the server of any changes.
+        /// </summary>
+        private void onSelectedModsChanged(ValueChangedEvent<IReadOnlyList<Mod>> mods)
+        {
+            // Process changes to the selection immediately because these occur somewhat infrequently.
+            flushPendingChanges();
+
+            modSettingChangeTracker?.Dispose();
+            modSettingChangeTracker = new ModSettingChangeTracker(mods.NewValue);
+            modSettingChangeTracker.SettingChanged += _ =>
+            {
+                // Debounce changes to mod settings because these can be continuous (e.g. moving a slider bar).
+                scheduledFlush ??= Scheduler.AddDelayed(flushPendingChanges, flush_debounce_time);
+            };
+        }
+
+        /// <summary>
+        /// Notifies the server of changes to the selected mods.
+        /// </summary>
+        private void flushPendingChanges()
+        {
+            scheduledFlush?.Cancel();
+            scheduledFlush = null;
+
+            if (client.Room == null)
+                return;
+
+            countPendingChanges++;
+
+            client.ChangeUserMods(SelectedMods.Value)
+                  .FireAndForget(endChange, _ => endChange());
```

Comment:
In the previous PR, I changed the return of `FireAndForget` to a `Task`, resulting in this comment: https://github.com/ppy/osu/pull/32250#discussion_r1998371184

I disagree with the reasoning - the fact that we have callbacks at all already makes it definitionally _not_ "fire and forget", but I suppose this version of binding both callbacks isn't too bad.

Reply:
Just to clarify: `FireAndForget()` is currently predominantly used wherever there is a a need to reconcile an async flow with sync code that is not easily changed. Stuff like drawable `Action`s, screen `OnExiting()` overrides, etc. The point is that you can call that in a sync flow to handle a `Task` and not worry about whether stuff gets awaited properly.

I was opposed to changing that method to return a `Task` because now the "async escape hatch" is not a hatch. Like what am I supposed to do with the `Task` that I just got *from the method that I use to escape `Task`s*? If we're going to go that way, then let's just delete the method and replace it with manual continuations everywhere.